addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -2194,3 +2194,35 @@ def test_multicompany_transit_with_one_company_for_user(self):
         intercom_location = self.env.ref('stock.stock_location_inter_company')
         self.assertEqual(so.picking_ids.location_dest_id, intercom_location)
         self.assertEqual(so.picking_ids.move_ids.location_dest_id, intercom_location)
+
+    def test_cancelled_backorder_product_forecast_widget(self):
+        '''
+        Ensure the availability widget is hidden if back order was cancelled.
```

Comment:
Ensure the availability widget is visible when the sale order line is linked to a picking, and hidden if the related moves are all cancelled or done.